spec/lib/saml/post_url_service_spec.rb

```diff
@@ -613,8 +613,27 @@
             let(:expected_log_message) { 'Redirecting to /terms-of-use' }
             let(:expected_log_payload) { { type: :ssoe } }
 
+            shared_examples 'client list' do |env, list|
+              before { allow(Settings).to receive(:vsp_environment).and_return(env) }
+
+              it 'returns the appropriate list of clients depending on the environment' do
+                result = subject.send(:enabled_tou_clients)
+                expect(result)
+                  .to eq(list)
+              end
+            end
+
+            context 'when the environment is not production' do
+              include_examples 'client list', ['production', SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS]
+            end
+
+            context 'when the environment is production' do
+              include_examples 'client list',
+                               ['some-environment', SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS_STAGING]
+            end
```

Comment:
It looks like these are switched around. The first one `is production` and the `second is not production`.